cdc/sink/mysql/mysql.go

```diff
@@ -452,10 +462,23 @@ func (s *mysqlSink) notifyAndWaitExec(ctx context.Context) {
 	s.broadcastFinishTxn()
 	s.execWaitNotifier.Notify()
 	done := make(chan struct{})
+
+	// spawn goroutine from s.workerWg to avoid goroutine leak when mysqlSink is closed.
+	s.workerWg.Add(1)
 	go func() {
-		s.flushSyncWg.Wait()
-		close(done)
+		defer s.workerWg.Done()
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				s.flushSyncWg.Wait()
```

Comment:
In the scenario we analyzed earlier, this goroutine will still be stuck here.

> When mysqlSink.Close() is called, all mysqlWorker will exit due to the ctx is canceled. At the same time if [this line](https://github.com/pingcap/tiflow/pull/5400/files#diff-ba27cb35a4f83fcd923c1403ff1fb7c1ad29659e7eadd87dd505700bf82f92b9R462) is excuted, the flushSyncWg.Add(1) was called but a corespondent flushSyncWg.Done() will never be called since all mysqlWorker was exited. So [this goroutine](https://github.com/pingcap/tiflow/pull/5400/files#diff-ba27cb35a4f83fcd923c1403ff1fb7c1ad29659e7eadd87dd505700bf82f92b9R468) will wait for a never done flushSyncWg.